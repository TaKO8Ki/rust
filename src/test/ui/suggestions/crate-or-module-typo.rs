// edition:2018

#![feature(decl_macro)]

macro a() {
    extern crate core as my_core;
    use my_cor::mem;
    mod a {
        pub fn bar() {}
    }
}

macro_rules! b {
    () => {
        mod b {
            pub fn bar() {}
        }
    }
}

mod foo {
    pub fn bar() { fooo::baz(); } //~ ERROR failed to resolve: use of undeclared crate or module `fooo`

    fn baz() {}
}

a!();

b!();

use my_cor::mem;

use my_core::mem;

use aa::bar; //~ ERROR unresolved import `aa`

use bb::bar; //~ ERROR unresolved import `bb`

use st::cell::Cell; //~ ERROR failed to resolve: use of undeclared crate or module `st`

use fooo::bar; //~ ERROR unresolved import `fooo`

struct Foo {
    bar: st::cell::Cell<bool> //~ ERROR failed to resolve: use of undeclared crate or module `st`
}

fn main() {}
