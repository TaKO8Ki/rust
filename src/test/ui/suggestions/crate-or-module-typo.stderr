error[E0432]: unresolved import `my_cor`
  --> $DIR/crate-or-module-typo.rs:31:5
   |
LL | use my_cor::mem;
   |     ^^^^^^ use of undeclared crate or module `my_cor`

error[E0432]: unresolved import `my_core`
  --> $DIR/crate-or-module-typo.rs:33:5
   |
LL | use my_core::mem;
   |     ^^^^^^^ use of undeclared crate or module `my_core`

error[E0432]: unresolved import `aa`
  --> $DIR/crate-or-module-typo.rs:35:5
   |
LL | use aa::bar;
   |     ^^ use of undeclared crate or module `aa`

error[E0433]: failed to resolve: use of undeclared crate or module `st`
  --> $DIR/crate-or-module-typo.rs:39:5
   |
LL | use st::cell::Cell;
   |     ^^ use of undeclared crate or module `st`
   |
help: there is a crate or module with a similar name
   |
LL | use std::cell::Cell;
   |     ~~~

error[E0432]: unresolved import `bb`
  --> $DIR/crate-or-module-typo.rs:37:5
   |
LL | use bb::bar;
   |     ^^ use of undeclared crate or module `bb`
   |
help: there is a crate or module with a similar name
   |
LL | use b::bar;
   |     ~

error[E0432]: unresolved import `fooo`
  --> $DIR/crate-or-module-typo.rs:41:5
   |
LL | use fooo::bar;
   |     ^^^^ use of undeclared crate or module `fooo`
   |
help: there is a crate or module with a similar name
   |
LL | use foo::bar;
   |     ~~~

error[E0432]: unresolved import `my_cor`
  --> $DIR/crate-or-module-typo.rs:7:9
   |
LL |     use my_cor::mem;
   |         ^^^^^^ use of undeclared crate or module `my_cor`
...
LL | a!();
   | ----- in this macro invocation
   |
   = note: this error originates in the macro `a` (in Nightly builds, run with -Z macro-backtrace for more info)
help: there is a crate or module with a similar name
   |
LL |     use my_core::mem;
   |         ~~~~~~~

error[E0433]: failed to resolve: use of undeclared crate or module `fooo`
  --> $DIR/crate-or-module-typo.rs:22:20
   |
LL |     pub fn bar() { fooo::baz(); }
   |                    ^^^^ use of undeclared crate or module `fooo`

error[E0433]: failed to resolve: use of undeclared crate or module `st`
  --> $DIR/crate-or-module-typo.rs:44:10
   |
LL |     bar: st::cell::Cell<bool>
   |          ^^ use of undeclared crate or module `st`
   |
help: there is a crate or module with a similar name
   |
LL |     bar: std::cell::Cell<bool>
   |          ~~~

error: aborting due to 9 previous errors

Some errors have detailed explanations: E0432, E0433.
For more information about an error, try `rustc --explain E0432`.
